pkg/controller/machinepool/nutanixactuator_test.go

```diff
@@ -0,0 +1,277 @@
+
+			actuator, err := NewNutanixActuator(getMasterMachineWithImage(t))
+			require.NoError(t, err, "unexpected error creating NutanixActuator")
+
+			_, _, err = actuator.GenerateMachineSets(test.clusterDeployment, test.pool, logger)
+
+			if test.expectedErr {
+				assert.Error(t, err, "expected error for test case")
+			} else {
+				assert.NoError(t, err, "unexpected error for test case")
+			}
+		})
+	}
+}
+
+func testNutanixPool() *hivev1.MachinePool {
+	return &hivev1.MachinePool{
+		Spec: hivev1.MachinePoolSpec{
+			Platform: hivev1.MachinePoolPlatform{
+				Nutanix: &hivev1nutanix.MachinePool{
+					NumCPUs:           4,
+					NumCoresPerSocket: 2,
+					MemoryMiB:         8192,
+				},
+			},
+		},
+	}
+}
+
+func testNutanixClusterDeployment() *hivev1.ClusterDeployment {
+	return &hivev1.ClusterDeployment{
+		Spec: hivev1.ClusterDeploymentSpec{
+			ClusterMetadata: &hivev1.ClusterMetadata{InfraID: "test-infra"},
+			Platform: hivev1.Platform{
+				Nutanix: &hivev1nutanix.Platform{
+					FailureDomains: []hivev1nutanix.FailureDomain{
+						{
+							Name: "NAME",
+							PrismElement: hivev1nutanix.PrismElement{
+								UUID: "test-prism-element-uuid",
+								Endpoint: hivev1nutanix.PrismEndpoint{
+									Address: "prism.example.com",
+									Port:    9440,
+								},
+							},
+							SubnetUUIDs:       nil,
+							StorageContainers: nil,
+							DataSourceImages:  nil,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestDecodeNutanixMachineProviderSpec(t *testing.T) {
+	t.Run("Valid provider spec", func(t *testing.T) {
+		providerConfig := &machinev1.NutanixMachineProviderConfig{
+			Image: machinev1.NutanixResourceIdentifier{Name: ptr("nutanix-image")},
+		}
+		data, err := json.Marshal(providerConfig)
+		assert.NoError(t, err)
+
+		rawExt := &runtime.RawExtension{Raw: data}
+		decodedSpec, err := decodeNutanixMachineProviderSpec(rawExt)
+		assert.NoError(t, err)
+		assert.NotNil(t, decodedSpec)
+		assert.Equal(t, "nutanix-image", *decodedSpec.Image.Name)
+	})
+
+	t.Run("Nil provider spec", func(t *testing.T) {
+		decodedSpec, err := decodeNutanixMachineProviderSpec(nil)
+		assert.NoError(t, err)
+		assert.NotNil(t, decodedSpec)
+	})
+}
+
+func TestGetRHCOSImageNameFromMasterMachine(t *testing.T) {
+	logger := log.New()
+	cd := &hivev1.ClusterDeployment{Spec: hivev1.ClusterDeploymentSpec{Platform: hivev1.Platform{}}}
+
+	t.Run("Valid RHCOS image name", func(t *testing.T) {
+		masterMachine := getMasterMachineWithImage(t)
+		imageName, err := getRHCOSImageNameFromMasterMachine(masterMachine, cd, logger)
+		assert.NoError(t, err)
+		assert.Equal(t, "nutanix-image-name", imageName)
+	})
+
+	t.Run("Missing RHCOS image name", func(t *testing.T) {
+		masterMachine := testMachine("master0", "master")
+		imageName, err := getRHCOSImageNameFromMasterMachine(masterMachine, cd, logger)
+		assert.Error(t, err)
+		assert.Empty(t, imageName)
+	})
+}
+
+func ptr(s string) *string {
```

Comment:
Or `To()` from `k8s.io/utils/ptr`. But it hardly seems worth a whole import :shrug:

Reply:
I thought that too, anyway, I ended up using this `ptr` method only once so I'll just do it the test itself.